Adding OOP tutorial blog post (#132)
NCAR/esds: posts/2021/oop-tutorial.md

---
author: Julia Kent
date: 2021-11-08
tags: python-tutorial-series
---

# Object Oriented Programming Tutorial

The NCAR/UCAR virtual Python Tutorial Seminar Series continues with a lesson on Object Oriented Programming on **November 10th at 1 PM Mountain Standard Time** led by Dr. Kevin Paul.

The content for this seminar tutorial is hosted on Kevin's [Object Oriented Programming Tutorial Repository](https://github.com/kmpaul/pg2oop).

## Bio

Dr. Kevin Paul is a computational physicist and software engineer working in the Computational & Information Systems Lab at the National Center for Atmospheric Research. Dr. Paul is the head of the Experimental Development Team (Xdev) and is passionate about open source development and teaching Python to Earth system scientists.

## How-to-Run

**This tutorial is designed to be launched from Binder.**

- Click the "Launch Binder" button at the top level of the [repository](https://github.com/kmpaul/pg2oop).

> 
> - Then, in the recently-launched Binder tab, select the notebook file `index.ipynb` to begin.

Reviewer: You don't need this step.  You only need this step if you run locally from your laptop.

```suggestion
```

> You may also download the environment to run locally:

Reviewer: ```suggestion
You may also download the tutorial from GitHub to run it locally:
```

- Check your conda install with `conda --version` and if necessary [install by following these instructions](https://docs.conda.io/en/latest/miniconda.html).

- Clone this repository by running `git clone https://github.com/kmpaul/pg2oop` in your terminal.

> - Still in the terminal, create a Conda environment with Jupyter installed by first moving into the `pg2oop` directory with `cd pg2oop` and then `conda create --name pg2oop jupyterlab`
> 

Reviewer: It may be less confusing here to split this into two steps:

```suggestion
- Still in the terminal, create a Conda environment with Jupyter installed by executing `conda create --name pg2oop jupyterlab`

- Still in the terminal, move into the `pg2oop` directory with `cd pg2oop`

```

- Activate your environment in the terminal with `conda activate pg2oop`.

> - Test your installation by launching JupyterLab with `jupyter lab`.

Reviewer: ```suggestion
- Launch the tutorial with JupyterLab by typing `jupyter lab` and then clicking on the `index.ipynb` file in Jupyter Lab's file browser.
```
